megatron/model/transformer.py

```diff
@@ -365,36 +370,44 @@ def forward(self, q, k, v):
         """
 
         assert all((i.dtype in [torch.float16, torch.bfloat16] for i in (q,k,v)))
-        assert all((i.is_cuda for i in (q,k,v)))
+        assert all((i.is_cuda or i.is_xpu for i in (q,k,v)))
```

Comment:
I feel it might be better to separate this into two assertions based on the value of "get_accelerator().device_name()", so that we only "assert all((i.is_cuda for i in (q,k,v)))" when "get_accelerator().device_name() == 'cuda'", and vice versa.